File: app/src/main/java/com/datarockets/mnchkn/fragments/players/PlayerInteractor.java
package com.datarockets.mnchkn.fragments.players;

public interface PlayerInteractor {
    interface OnUpdatedListener {
        void onUpdateLevel(int score);
        void onUpdateStrength(int score);
    }
    void increaseLevelScore(int score, OnUpdatedListener listener);
    void decreaseLevelScore(int score, OnUpdatedListener listener);
    void increaseStrengthScore(int score, OnUpdatedListener listener);
    void decreaseStrengthScore(int score, OnUpdatedListener listener);
}

File: app/src/main/kotlin/com/datarockets/mnchkn/fragments/players/PlayerInteractor.kt
package com.datarockets.mnchkn.fragments.players

interface PlayerInteractor {
    interface OnUpdatedListener {
        fun onUpdateLevel(score: Int)
        fun onUpdateStrength(score: Int)
    }

    fun increaseLevelScore(score: Int, listener: OnUpdatedListener)
    fun decreaseLevelScore(score: Int, listener: OnUpdatedListener)
    fun increaseStrengthScore(score: Int, listener: OnUpdatedListener)
    fun decreaseStrengthScore(score: Int, listener: OnUpdatedListener)
}
